syntax = "proto3";

package sound;

message SoundGenConfig {
  int64 fs = 1;  // sampling frequency (Hz)
  int64 duration = 2;  // (sec)

  oneof type {
    PureTone pure_tone_config = 3;
    FlatSpectrumNoise flat_spectrum_noise_config = 4;
    Chirp chirp_config = 5;
  }
}

message PureTone {
  int64 center_freq = 1;
}

message Chirp {
  int64 start_freq = 1;
  int64 stop_freq = 2;
  enum SweepMethod {
    LINEAR = 0;
    QUADRATIC = 1;
    LOGARITHMIC = 2;
    HYPERBOLIC = 3;
  }
  SweepMethod sweep_method = 3;
}

message FlatSpectrumNoise {
  int64 filter_order = 3;  // Order of butterworth filter used to bandpass noise
  int64 start_freq = 1;
  int64 stop_freq = 2;
}

